### PR TITLE
src: build fix for Android

### PR DESCRIPTION
**Pull request by devnexen**
pipe2 needs GNU source defined.

### CODE REVIEW

**Comment by bnoordhuis** [comment]
LGTM but the more principled approach is to define it in CMakeLists.txt and Makefile.am (grep for Android and ANDROID respectively.)

Libuv defines `_GNU_SOURCE` in one or two source files but it should be safe to remove those.

**Comment by gengjiawen** [comment]
Will this eliminate those two warnings (only those two on android platform) ? 

```console
/__w/libuv/libuv/src/unix/process.c:144:7: warning: implicit declaration of function 'pipe2' [-Wimplicit-function-declaration]
  if (pipe2(fds, flags | O_CLOEXEC))
      ^
/__w/libuv/libuv/src/unix/process.c:144:7: warning: this function declaration is not a prototype [-Wstrict-prototypes]
```

https://github.com/libuv/libuv/pull/2858/checks?check_run_id=692991457#step:5:76

File: src/unix/process.c [hunk 1]
```diff
@@ -19,6 +19,7 @@
  * IN THE SOFTWARE.
  */
 
+#define _GNU_SOURCE
```
Comment (gengjiawen) [inline src/unix/process.c:22]: Should it be like in `src\unix\linux-syscalls.h`
```c
#undef  _GNU_SOURCE
#define _GNU_SOURCE
```

**Comment by gengjiawen** [comment]
> LGTM. It might be nice to wait for #2858 to land first.

#2858 had landed, can you rebase this pr ?

File: src/unix/process.c [hunk 2]
```diff
@@ -19,6 +19,9 @@
  * IN THE SOFTWARE.
  */
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
```
Comment (bnoordhuis) [inline src/unix/process.c:24]:
```suggestion
```
(not necessary anymore)

**Comment by bnoordhuis** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/1932/

**Comment by bnoordhuis** [comment]
Sorry, forgot to merge this. Landed in 40bab5ed, thanks David.

**Pull request closed without merging**